torchci/lib/bot/retryBot.ts

```diff
@@ -54,27 +53,43 @@ function retryBot(app: Probot): void {
       // or a widespread outage that wouldn't be helped by retries
       return;
     }
+
+    // @ts-expect-error - we don't have types for these
+    let doesLookLikeInfraFailure = (job, isInfraStep: (step: any) => boolean) => {
+      // Ensure that the steps that failed are only infra related steps (e.g. they're not lint, build, test steps)
+      return job.steps?.filter(
+        // @ts-expect-error
+        (step) =>
+          step.conclusion !== null &&
```

Comment:
Q: If step before non-retriable one in the job has failed (with infra failure), what would be the conclusion of the next step? If it will be failure as well, then the logic would not work, would it? Can we add a unit test of sorts for this one (using a simulated failure in canary or ones personal repo?)

Reply:
I believe that it’s status is null, though I can’t remember a repro at the moment. 

But we can fix that later if this belief turns out to be wrong (at worst, it'll not retry something that we'd like it to retry)

Reply:
Well, it would be good to have an example on PR description, where it works as expected, otherwise it's not really useful (i.e. if later API will change it would be good to have a reference to a runs when it was working as expected)
And in general, testing code before landing is a good idea.